palette/src/equality.rs

```diff
@@ -1,38 +1,61 @@
 use num_traits::Float;
-use approx::ApproxEq;
+
+use approx::{AbsDiffEq, RelativeEq, UlpsEq};
 
 use {cast, Component, Lab, LabHue, Lch, RgbHue, Xyz, Yxy};
 use white_point::WhitePoint;
 
 macro_rules! impl_eq {
     (  $self_ty: ident , [$($element: ident),+]) => {
-        impl<Wp, T> ApproxEq for $self_ty<Wp, T>
-        where T: Component + Float + ApproxEq,
+        impl<Wp, T> AbsDiffEq for $self_ty<Wp, T>
+        where T: Component + Float + AbsDiffEq,
             T::Epsilon: Copy + Float,
-            Wp: WhitePoint
+            Wp: WhitePoint + PartialEq
         {
-            type Epsilon = <T as ApproxEq>::Epsilon;
+            type Epsilon = T::Epsilon;
 
             fn default_epsilon() -> Self::Epsilon {
                 T::default_epsilon()
             }
-            fn default_max_relative() -> Self::Epsilon {
-                T::default_max_relative()
+
+            fn abs_diff_eq(&self, other: &Self, epsilon: T::Epsilon) -> bool {
+                $( self.$element.abs_diff_eq(&other.$element, epsilon) )&&+
             }
-            fn default_max_ulps() -> u32 {
-                T::default_max_ulps()
+            fn abs_diff_ne(&self, other: &Self, epsilon: T::Epsilon) -> bool {
+                $( self.$element.abs_diff_ne(&other.$element, epsilon) )&&+
```

Comment:
I think these should be delimited with `||`, since it would be the inverse of `abs_diff_eq`. Same for the other `*_ne` implementations.

Reply:
You're right that should be `||`, looks it was implemented wrong previously as well because the `*_ne` were using `&&`. I just adapted that by accident

Reply:
Oops, good thing it was discovered then.